File: onlineStoreExample.java
import java.util.HashMap;
import java.util.Map;
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class ZiftMedia {

	public static void main(String[] args) {

		System.out.println("Welcome! Today we are having our Independence Day Sale!  All CDs are 50% off!  All DVDs are 25% off! AND All BLU-RAYs are 15% off!  PLUS...Buy 5 or more DVDs and they are 30% off!");

    // build list of store items and create cost and discount hashes (versus loading database)
    List<String> storeItems = new ArrayList<String>();
    storeItems.add("CD");
    storeItems.add("DVD");
    storeItems.add("BLU-RAY");
    Map<String, Double> costPerUnit = new HashMap<String, Double>();
		costPerUnit.put("CD", 9.99);
		costPerUnit.put("DVD", 12.99);
		costPerUnit.put("BLU-RAY", 15.99);
    Map<String, Double> discountRates = new HashMap<String, Double>();
		discountRates.put("CD", 0.50);
		discountRates.put("DVD", 0.25);
		discountRates.put("BLU-RAY", 0.15);

    // initialize pre-discounted total cost and total savings variables
		Double totalCost = 0.00;
		Double totalSavings = 0.00;

    // loop through all store items
		for (int i = 0; i < storeItems.size(); i++) {
			String item = storeItems.get(i);

			// accept quantity as input from user
			System.out.println("How many " + item + "s would you like to buy today?");
			Scanner input = new Scanner(System.in);
    			int quantity = input.nextInt();

  		// add associated cost per item to total cost
  		totalCost = totalCost + costPerUnit.get(item) * quantity;

  		// add associated savings per item to total savings
  		// account for DVD quantity-driven discount factor
  		if (item == "DVD" && quantity >= 5) {
  			totalSavings = totalSavings + costPerUnit.get(item) * quantity * 0.30;
  		} else {
  			totalSavings = totalSavings + costPerUnit.get(item) * quantity * discountRates.get(item);
  		}

		}

		// calculate final total and print to screen
		Double finalOutput = Math.round((totalCost - totalSavings) * 100.0) / 100.0;
		System.out.print("Your total today is $" + finalOutput + ".");

	}

}

File: onlineStoreExample.rb
# Assume that there is an unlimited inventory of digital media.
# Propose that each music CD has a base price_per_unit of $9.99
# Propose that each movie DVD has a base price_per_unit of $12.99
# Propose that each movie Blu-ray has a base price_per_unit of $15.99

store_options = ["CD", "DVD", "BLU-RAY"]
cost_per_unit = {"CD"=>9.99, "DVD"=>12.99, "BLU-RAY"=>15.99}
discount_rates = {"CD"=>0.5, "DVD"=>0.25, "BLU-RAY"=>0.15}
total_cost = 0
total_savings = 0

puts "Welcome! Today we are having an Independence Day Sale!\nAll CDs are 50% off!  All DVDs are 25% off! All BLU-RAYs are 15% off!\nPLUS...Buy 5 or more DVDs and they are 30% off!\n\n"


store_options.each do |media|
  puts "How many #{media}'s would you like to buy today?"
  quantity = gets.chomp.to_i  # Assumes 0 if given non-number entry
  total_cost += cost_per_unit[media] * quantity
  if media == "DVD" && quantity >= 5
    total_savings += (cost_per_unit[media] * quantity * 0.3)
  else
    total_savings += (cost_per_unit[media] * quantity * discount_rates[media])
  end
end

puts "Your total today is $#{(total_cost - total_savings).round(2)}."
